scio-elasticsearch/es6/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -367,30 +365,42 @@ public PDone expand(final PCollection<T> input) {
         checkArgument(maxBulkRequestBytes > 0);
         checkArgument(maxRetries >= 0);
         checkArgument(retryPause.getMillis() >= 0);
-        input
-            .apply("Assign To Shard", ParDo.of(new AssignToShard<>(numOfShard)))
-            .apply(
-                "Re-Window to Global Window",
-                Window.<KV<Long, T>>into(new GlobalWindows())
-                    .triggering(
-                        Repeatedly.forever(
-                            AfterProcessingTime.pastFirstElementInPane()
-                                .plusDelayOf(flushInterval)))
-                    .discardingFiredPanes()
-                    .withTimestampCombiner(TimestampCombiner.END_OF_WINDOW))
-            .apply(GroupByKey.create())
-            .apply(
-                "Write to Elasticsearch",
-                ParDo.of(
-                    new ElasticsearchWriter<>(
-                        clusterName,
-                        servers,
-                        maxBulkRequestSize,
-                        maxBulkRequestBytes,
-                        toDocWriteRequests,
-                        error,
-                        maxRetries,
-                        retryPause)));
+        if (numOfShard == 1) {
```

Comment:
Should this be `0` instead? If `numOfShard == 1` we explicitly want to shuffle everything into a single shard? `0` is also consistent with file-based IOs, i.e. do not reshuffle, write each bundle as is.

Reply:
Addressed.